File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if(len(nums) < 1):
            return 0
        elif(len(nums) < 2):
            return 1

        nums.sort()
        i = 0
        j = i+1
        temp = 1
        cs = 0
        while(i < len(nums)-1):
            if(nums[j]-1 == nums[i]):
                temp += 1
                i += 1
                j += 1
            elif(nums[j] == nums[i]):
                i += 1
                j += 1
                continue
            else:
                cs = max(temp, cs)
                temp = 1
                i += 1
                j += 1
        return max(temp, cs)

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int[] prefix = new int[nums.length];
        int sum = 0;
        for(int i = 0; i<prefix.length; i++){
            sum+=nums[i];
            prefix[i] = sum;
        }

        int c = 0;
        for(int i = 0; i<prefix.length; i++){
            if(prefix[i] == k) c++;
            for(int j = i+1; j < prefix.length; j++){
                if(prefix[j] - prefix[i] == k) c++;
            }
        }
        
        return c;
    }
}